src/renderers/shared/server/ReactPartialRenderer.js

```diff
@@ -320,6 +321,7 @@ function validateRenderResult(child, type) {
 
 function resolve(child, context) {
   while (React.isValidElement(child)) {
+    child = (child: React.Element);
```

Comment:
Why is this needed?

Reply:
I’m not really sure why Flow isn’t understanding that `child` is a `React.Element` at this point without this extra hint. I thought that by typing the callsites to `resolve` this would just work.